core/src/main/java/org/commonjava/maven/ext/core/impl/DependencyManipulator.java

```diff
@@ -626,19 +633,25 @@ else if ( strict && oldVersion.contains( "$" ) &&
      * Remove version overrides which refer to projects in the current reactor.
      * Projects in the reactor include things like inter-module dependencies
      * which should never be overridden.
+     *
      * @param versionOverrides current set of ArtifactRef:newVersion overrides.
      * @return A new Map with the reactor GAs removed.
      */
     private Map<ArtifactRef, String> removeReactorGAs( final Map<ArtifactRef, String> versionOverrides )
     {
         final Map<ArtifactRef, String> reducedVersionOverrides = new LinkedHashMap<>( versionOverrides );
-        Iterator<Entry<ArtifactRef, String>> it = reducedVersionOverrides.entrySet().iterator();
+        final Iterator<Entry<ArtifactRef, String>> it = reducedVersionOverrides.entrySet().iterator();
         while ( it.hasNext() )
         {
-            Entry<ArtifactRef, String> e = it.next();
+            final Entry<ArtifactRef, String> e = it.next();
+
             session.getProjects().forEach( p -> {
-                if ( e.getKey().asProjectVersionRef().equals( new SimpleProjectVersionRef( p.getGroupId(), p.getArtifactId(), p.getVersion() ) ) )
+                final ProjectRef currentOverride = e.getKey().asProjectRef();
+                final ProjectRef projectGA = new SimpleProjectRef( p.getGroupId(), p.getArtifactId() );
+
+                if ( currentOverride.equals( projectGA ) )
```

Comment:
Thinking on this we could probably make it simpler by just doing:
```
if ( e.getKey().getGroupId().equals(p.getGroupId()) && e.getKey().getArtifactId().equals(p.getArtifactId() )
{
   logger.warn....
```

This avoids creating the new objects and is a bit simpler.

Reply:
Done in 90013dc.